code/scripting/api/objs/hudconfig.cpp

```diff
@@ -52,6 +52,86 @@ bool hud_preset_h::isValid() const
 	return preset >= 0 && preset < (int)HC_preset_filenames.size();
 }
 
+hud_color_preset_h::hud_color_preset_h() : preset(-1) {}
+hud_color_preset_h::hud_color_preset_h(int l_preset) : preset(l_preset) {}
+
+int hud_color_preset_h::getIndex() const
+{
+	return preset;
+}
+
+SCP_string hud_color_preset_h::getName() const
+{
+	if (!isValid()) {
+		return "";
+	}
+
+	return XSTR(HC_colors[preset].name.c_str(), HC_colors[preset].xstr);
+}
+
+bool hud_color_preset_h::isValid() const
+{
+	return preset >= 0 && preset < (int)HC_preset_filenames.size();
+}
```

Comment:
Static cast?

Reply:
honestly surprised the automated checks didn't enforce that... Done. Also did it for where I copied it from.